sections/method/preprocessing.tex

```diff
@@ -0,0 +1,49 @@
+\subsection{Preprocessing}\label{subsec:prepro}
+
+Our preprocessing consists of several steps, each of which will be described in more detail.
+
+\begin{enumerate}
+	\item Article Extraction
+	\item Initial Document filtering
+	\item Stopwords and Word Frequency
+	\item Word Databases / POS Tagging
+	\item Stemming
+	\item Repeat step 2 and Stopwords
+	\item Construct Files
+\end{enumerate}
+
+\paragraph{Article Extraction}
+Documents are loaded from \texttt{.xml} files, and their title and body contents extracted and converted into \texttt{.json} format.
+
+\paragraph{Initial Document Filtering}
+Duplicate documents are removed. Documents containing fewer than 20 words are also removed.
+
+\paragraph{Stopwords and Word Frequncy Filtering}
+Stopwords are removed, using \emph{nltk.corpus.stopwords}. 
+Words are removed if they appear fewer than 25 times in the document set or if they are used in over 25\% of the documents.
+
+\paragraph{Words Database Filtering / POS-Tagging}
+Words are checked against the two word databases: Wiktionary and DanNet. 
+The databases contain only: nouns, adjectives, verbs, adverbs, and proper nouns. 
+Words that are not contained within either database are removed.
+
+\paragraph{Stemming}
+Words are reduced to their stems using \texttt{nltk.stem.snowball.DanishStemmer}. 
+This is done to combine word declensions, which reduced the number of words without removing too much information.
+Words that are stemmed into stopwords are also removed at this time.
+
+\paragraph{Construct files}
+Construct and save scikit-learn.CountVectorizer as a sparse matrix \texttt{.npz} file, and word2vec, doc2vec, doc2word as \texttt{.csv} files.
+
+
+\subsection{Post-LDA preprocessing}
+After LDA has made a topic model, documents which have no accociated topic, and topics which have none accociated documents would be removed.
+Topics that occur in almost every document are also considered too general, and would also be removed.
+Initially, this presented some implementation challenges, as it required removing topics and documents from files generated earlier in the process and from the LDA model, to be able to use the right topic/document IDs in the future.
+
+\subsection{Query Preprocessing}
+One of the challenges of preprocessing queries is that they usually contain very few words, meaning that there is a huge possibility of information loss during preprocessing and conversion, as there is not much content or context to be learned from.
+
+Our method converts each word in the query into a topic distribution, making the query have the same format as a document. 
```

Comment:
Passer det? Det lyder lidt mystisk, da hvert ord er givet et specifikt topic, så hvordan skaffes der en distribution? Hvis det passer som det står, skal der uddybes hvor distributionen kommer fra.

Reply:
så vidt jeg ved så ender hvert ord faktisk med at kunne give flere topics (syntes vi har erfaret at flere topics bruger de samme ord). men derudover tror jeg bare at man ville dele den op per ord, så en query med to ord ville have 50/50 distribution på to topics.
Det var ihvertfald min tanke.

Reply:
Ok. Jeg troede at LDA specifikt gav et topic til hvert ord, men jeg kan tage fejl. Jeg tror din idé er fin nok, men lad os sætte en todo, indtil vi er sikre på hvordan det bliver gjort.
```suggestion
Our method converts each word in the query into a topic distribution, making the query have the same format as a document. \todo[inline]{Detaljer specifikt hvordan denne distribution laves, når det er færdigimplementeret.}
```